src/main/java/org/broadinstitute/consent/http/resources/DaaResource.java

```diff
@@ -83,6 +87,46 @@ public Response createDaaForDac(
     }
   }
 
+  @PUT
+  @Produces(MediaType.APPLICATION_JSON)
+  @RolesAllowed({ADMIN, SIGNINGOFFICIAL})
+  @Path("{daaId}/{userId}")
+  public Response createLibraryCardDaaRelation(
+      @Context UriInfo info,
+      @Auth AuthUser authUser,
+      @PathParam("daaId") Integer daaId,
+      @PathParam("userId") Integer userId) {
+    try {
+      User authedUser = userService.findUserByEmail(authUser.getEmail());
+      int authedUserInstitutionId = authedUser.getInstitutionId();
+      User user = userService.findUserById(userId);
+      int userInstitutionId = user.getInstitutionId();
+      // Assert that the user has the correct institution permissions to add a DAA-LC relationship.
+      // Admins can add a DAA with any DAC, but signing officials can only create relationships for
+      // library cards associated with the same institution they are associated with.
+      if (!authedUser.hasUserRole(UserRoles.ADMIN)) {
+        if (authedUserInstitutionId != userInstitutionId) {
+          return Response.status(Status.FORBIDDEN).build();
+        }
+      }
+      List<LibraryCard> libraryCards = libraryCardService.findLibraryCardsByUserId(userId);
+      Optional<LibraryCard> matchingCard = libraryCards.stream()
+          .filter(card -> card.getInstitutionId() == authedUser.getInstitutionId().intValue())
+          .findFirst();
+      if (matchingCard.isEmpty()) {
+        return Response.status(Status.NOT_FOUND).build();
+      }
```

Comment:
Thinking out loud here ... one thing we could do here is to push this logic to the service class and in the case that there is no LC, we could create one. However... that means the OK response should really be a Created response ... so I don't think we should go down that route - let's keep the APIs distinct.